docs/sql/scalar.txt

```diff
@@ -835,9 +835,152 @@ Examples
     +---------------------+--------------+
     SELECT 5 rows in set (... sec)
 
+
+Array Functions
+===============
+
+array_cat(first_array, second_array, [ third_array , ... ]) returns array
+-------------------------------------------------------------------------
+
+The ``array_cat`` function concatenates one or more arrays into one array
+
+::
+
+    cr> select array_cat([1,2,3],[3,4,5,6],[7,9,10]) from sys.cluster;
+    +-----------------------------------------+
+    | array_cat([1,2,3], [3,4,5,6], [7,9,10]) |
+    +-----------------------------------------+
+    | [1, 2, 3, 3, 4, 5, 6, 7, 9, 10]         |
+    +-----------------------------------------+
+    SELECT 1 row in set (... sec)
+
+
+It can be used to append elements to array fields
+
+::
+
+    cr> create table array_cat_example (list array(integer));
+    CREATE OK (... sec)
+
+::
+
+    cr> insert into array_cat_example (list) values ([1,2,3]);
+    INSERT OK, 1 row affected (... sec)
+
+.. Hidden: refresh array_cat_example
+
+    cr> refresh table array_cat_example
+    REFRESH OK (... sec)
+
+::
+
+    cr> update array_cat_example set list = array_cat(list, cast([4, 5, 6] as array(integer)));
```

Comment:
In order to be more flexible and to avoid this cumbersome cast with literals the functions should accept arrays as second argument whose inner type is convertable to the inner type of the first argument (``DataType.isConvertableTo(DataType)``) in the Resolver. The functions should wrap the second argument in a Cast function for the first argument type in ``normalizeSymbol()``.